libs/api/domains/air-discount-scheme/src/lib/models/discount.model.ts

```diff
@@ -9,8 +9,8 @@ export class Discount {
   @Field(() => ID)
   nationalId!: string
 
-  @Field()
-  discountCode!: string
+  @Field(() => String, { nullable: true })
+  discountCode!: string | null
```

Comment:
"I assure you, it's never not there... except when it is :sweat_smile: "

Reply:
I thought `?` in the typing system meant `undefined`.
I'd have it either be string or null.
Am I mistaken? :sweat_smile:

Reply:
Alternatively, do you have suggestions on how to represent it better?

Reply:
Here  I was pointing out the contradiction between the [non-null operator](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#non-null-assertion-operator-postfix-) and typing as optional `null`.

If the type-checker doesn't complain and the tests covering this passes, then I don't really mind it.

Reply:
I made the default value null instead of typing with `?`
I also fixed the tests when I realized they were a bit off on what they represented.

Tests and an additional smoketest working as expected.

Thanks for the tip :+1: